### ISSUE TITLE
wait(fn) stops evaluating after a few cycles

### ISSUE THREAD

**Issue by FdezRomero** [segmentio/nightmare#251]
Hello there,

I have been trying to use `.wait(fn)` and my script always got stuck at this point, so I started debugging the `waitfn()` function in `lib/actions.js`, adding some `debug()` like this:

```javascript
function waitfn (self, fn, done) {
  debug('waitfn');
  self._evaluate(fn, function (err, result) {
    debug('_evaluate');
    if (result) {
      debug('Wait is done');
      return done();
    }
    else {
      debug('Waiting another 250ms');
      setTimeout(function () {
        waitfn(self, fn, done);
      }, 250);
    }
  });
}
```

And noticed that `waitfn()` stops iterating after a few cycles. This is the output:

```
  nightmare:actions .wait() for fn +26ms
  nightmare:actions waitfn +0ms
  nightmare:actions _evaluate +3ms
  nightmare:actions Waiting another 250ms +1ms
  nightmare:actions waitfn +254ms
  nightmare:actions _evaluate +2ms
  nightmare:actions Waiting another 250ms +1ms
  nightmare:actions waitfn +251ms
  nightmare:actions _evaluate +191ms
  nightmare:actions Waiting another 250ms +0ms
  nightmare:actions waitfn +252ms
  // missing a call to "_evaluate" here
```

Any idea why this is happening? `_evaluate` only gets called 2 or 3 times and I really don't know why...

Thanks,
Rodrigo

**Comment by tjokimie** [comment]
I'm also facing the exact same problem. `waitfn` stop iterating after 1-3 iterations. If I increase the timeout from `250` to 2500 or more the script executes successfully.

**Comment by CatTail** [comment]
Any progress on this issues?

**Comment by kazinov** [comment]
+1

**Comment by rosshinkley** [comment]
@FdezRomero @CatTail @kazinov @tjokimie I cannot reproduce this problem.  Has this issue been resolved?

**Comment by rosshinkley** [comment]
I'm going to consider this issue resolved.  If this is still a problem, feel free to reopen/submit a new issue.

**Comment by deffwe** [comment]
seems that the bug isn't resolved yet, i'm on version 2.10. anybody has solution? thx

**Comment by rosshinkley** [comment]
@deffwe If you have a minimum repro, post it and I'll reopen this issue to take a look.
